migrations/emailmonitor/oxa_email_monitor.py

```diff
@@ -0,0 +1,215 @@
+def process_error_line( cursor, line ):
+
+    # From the activation email error line, with regular expression fetch IP, year, month, day, hour, minute, second, and email    
+    m = re.search('/tmp/oxa_log_files/(.+?)/.+:([0-9]{4}?)-([0-9]{2}?)-([0-9]{2}?)\s([0-9]{2}?):([0-9]{2}?):([0-9]{2}?),.+to\s"(.+?)"',line)
+    if m:
+        ip = m.group(1)
+        y = m.group(2)
+        mo = m.group(3)
+        d = m.group(4)
+        h = m.group(5)
+        mi = m.group(6)
+        s = m.group(7)        
+        dt_obj = datetime(int(y),int(mo),int(d),int(h),int(mi),int(s))       
+        email = m.group(8)
+                
+	    # If this error line is not already processed (not in MySQL) , process it otherwise skip 		
+        cursor.execute("SELECT id from oxa.oxa_activationfailed where email='"+email+"'")
+        if cursor.rowcount <= 0: 
+		    # Find user_id from email 
+            cursor.execute("SELECT id,date_joined from edxapp.auth_user WHERE email='"+email+"'")
+            user_id = 0
+			# Here we are converting time to PST time by subtracting 7, Normally in log files it is UTC time. But in MySQL it is PST time. So we map them to eachother as PST time
+            date_joined = dt_obj + timedelta(hours=7)
+            date_failed = date_joined
+            #Find the activation ket 
+			activation_key = "Not Found!!!"
+            # Find activation key from user_id
+			if cursor.rowcount > 0:
+                result = cursor.fetchall()         
+                user_id = result[0][0]
+                date_joined = result[0][1] 
+                cursor.execute("SELECT activation_key FROM edxapp.auth_registration WHERE user_id="+str(user_id)) 
+                if cursor.rowcount > 0:
+                    result = cursor.fetchall()
+                    activation_key = result[0][0]
+            
+			# Insert this new activation email failure to oxa.oxa_activationfailed table and update activation_failed statistics for corresponding day in oxa.oxa_activationsummary table
+            cursor.execute("INSERT INTO oxa.oxa_activationfailed (xyear,xmonth,xday,user_id,email,activation_key,date_joined,date_failed,hostvmip) VALUES("+y+","+mo+","+d+","+str(user_id)+",'"+email+"','"+activation_key+"','"+str(date_joined)+"','"+str(date_failed)+"','"+ip+"')")
+            cursor.execute("UPDATE oxa.oxa_activationsummary SET failed=failed+1 WHERE  xyear="+y+" and xmonth="+mo+" and xday="+d)    
+             
+        
+
+    
+# Fetch the error log files for the specified IP VMs and process them in order to determine activation email failures
+def fetch_and_grep_log_files():
+    # Remove if exists and create the folder for fethcing error log files from edxapp VMs
+	os.system('rm -fr /tmp/oxa_log_files')
+    os.system('mkdir /tmp/oxa_log_files')
+
+	# For each VM, create folder with its IP and copy log files with scp 
+    vms = ('10.0.0.?','10.0.0.?','10.0.0.?','10.0.0.?','10.0.0.?')
```

Comment:
Let's submit this first PR. Later we can improve it if want to invest more on this script.